external/esp_idf_port/esp32/esp32_queue_api.c

```diff
@@ -0,0 +1,434 @@
+/******************************************************************
+ *
+ * Copyright 2019 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************/
+
+#include <stddef.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <assert.h>
+#include <semaphore.h>
+#include <pthread.h>
+#include <errno.h>
+#include <sched.h>
+#include <debug.h>
+#include <sys/time.h>
+#include <sys/types.h>
+#include <fcntl.h>
+#include <tinyara/arch.h>
+#include <tinyara/clock.h>
+#include "esp_attr.h"
+#include "esp32_queue_api.h"
+#include "mq_tryreceive.h"
+/************************************************************************
+ * Pre-processor Definitions
+ ************************************************************************/
+#define MAX_QUEUE_INFO 10
+
+/************************************************************************
+ * Private Type Declarations
+ ************************************************************************/
+enum {
+	NORMAL = 0,
+	MIDDLE,
+	HIGH,
+} queue_prio_e;
+
+/************************************************************************
+ * Public Variables
+ ************************************************************************/
+const char *mq_name = "mq_wifi";
+static queue_info_t queues_info[MAX_QUEUE_INFO];
+
+/************************************************************************
+ * Public Functions
+ ************************************************************************/
+/****************************************************************************
+ * Name: queue_create_wrapper
+ *
+ * Description:
+ *	This function is the wrapper of mq_open to establish a connection between a named
+ * message queue and the calling task.
+ *
+ * Inputs:
+ *	queue_len - The length of the queue.
+ *	item_size - The size of each node of the queue.
+ *
+ * Return:
+ *	Returns the pionter of the created mqueue, but NULL in a failure.
+ *
+ ****************************************************************************/
+void *IRAM_ATTR queue_create_wrapper(uint32_t queue_len, uint32_t item_size)
+{
+#define NAME_LEN 20
+	char name[NAME_LEN];
+	bool flag = false;
+	uint32_t mq_id = 0;
+
+	for (int i = 0; i < MAX_QUEUE_INFO; i++) {
+		if (!queues_info[i].valid) {
+			flag = true;
```

Comment:
variable flag is unnecessary , just init mq_id as -1, then check it for no queue case

Reply:
good idea,  use this logic can save memory. 
but i can not find proper place to do init func to set -1, 
place at  board boot?